src/common/types.cpp

```diff
@@ -22,6 +25,10 @@ LogicalType::LogicalType(LogicalTypeId id, shared_ptr<ExtraTypeInfo> type_info_p
 	physical_type_ = GetInternalType();
 }
 
+LogicalType::LogicalType(LogicalTypeId id, PhysicalType internal_type, shared_ptr<ExtraTypeInfo> type_info_p)
+    : id_(id), physical_type_(internal_type), type_info_(move(type_info_p)) {
```

Comment:
This constructor seems a bit dangerous, because this might mean `internal_type` does not match `GetInternalType()`. Do we need this?

Reply:
The problem here is on how to get the Physical Type of the enum. We can't really check the EnumExtraInfo since we don't know (if we don't have the physical type at that point), because we won't know the templated type of EnumExtraInfo.